pkg/controllers/aws_infra.go

```diff
@@ -37,8 +37,6 @@ import (
 func (r *HypershiftDeploymentReconciler) createAWSInfra(hyd *hypdeployment.HypershiftDeployment, providerSecret *corev1.Secret) (ctrl.Result, error) {
 
 	var iamOut *aws.CreateIAMOutput
-	oHyd := *hyd.DeepCopy()
```

Comment:
Replaced all the deep copies, with a Client.Get.  This makes sure the base resource is the latest.